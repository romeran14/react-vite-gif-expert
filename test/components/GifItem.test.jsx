import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('first', () => {
    const url ='https://upload.wikimedia.org/wikipedia/commons/a/ad/Museo_del_Bicentenario_-_Zapatos_de_N%C3%A9stor_Kirchner.jpg'
    const title = 'goku'

    test('debe de hacer match con el snapshot', () => {
        
        const { container } = render( <GifItem title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();

    });
    
    test('debe mostrar la imagen con la URL y ALT indicado', () => { 
        render( <GifItem title={title} url={url}/>)
       // screen.debug()
       //expect( screen.getAllByRole('img'))
       const { src } = screen.getByRole('img');
       expect( src ).toBe( url );
     })

     test('debe mostrar el titulo', () => { 
        render( <GifItem title={title} url={url}/>)
        expect( screen.getByText( title) ).toBeTruthy()
      })
})