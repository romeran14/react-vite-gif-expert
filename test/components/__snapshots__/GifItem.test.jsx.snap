// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="goku"
      src="https://upload.wikimedia.org/wikipedia/commons/a/ad/Museo_del_Bicentenario_-_Zapatos_de_N%C3%A9stor_Kirchner.jpg"
    />
    <p>
      goku
    </p>
  </div>
</div>
`;
